ai-engineering/evaluate/flags-experiments.mdx

```diff
@@ -0,0 +1,387 @@
+  --flag.ticketClassification.model=gpt-4o \
+  --flag.ticketClassification.temperature=0.3 \
+  --flag.ticketClassification.systemPrompt=detailed
+```
+
+### JSON configuration files
+
+For complex experiments, define flag overrides in JSON files:
+
+```json experiments/gpt4-detailed.json
+{
+  "ticketClassification": {
+    "model": "gpt-4o",
+    "temperature": 0.3,
+    "systemPrompt": "detailed"
+  }
+}
+```
+
+```json experiments/gpt4-mini-concise.json
+{
+  "ticketClassification": {
+    "model": "gpt-4o-mini",
+    "temperature": 0.7,
+    "systemPrompt": "concise"
+  }
+}
+```
+
+Run evaluations with these configurations:
+
+```bash
+# Run with first configuration
+axiom eval --flags-config=experiments/gpt4-detailed.json
+
+# Run with second configuration
+axiom eval --flags-config=experiments/gpt4-mini-concise.json
+```
+
+<Tip>
+Store experiment configurations in version control. This makes it easy to reproduce results and track which experiments you've tried.
+</Tip>
+
+### Comparing experiments
+
+Run the same evaluation with different flag values to compare approaches:
+
+```bash
+# Baseline: default flags (gpt-4o-mini, temp 0.7, concise)
+axiom eval spam-classification
+
+# Experiment 1: Try GPT-4o
+axiom eval spam-classification --flag.ticketClassification.model=gpt-4o
+
+# Experiment 2: Lower temperature for more consistency
+axiom eval spam-classification --flag.ticketClassification.temperature=0.2
+
+# Experiment 3: Detailed prompting
+axiom eval spam-classification --flag.ticketClassification.systemPrompt=detailed
+```
+
+Axiom tracks all these runs in the Console, making it easy to compare scores and identify the best configuration.
+
+## Best practices
+
+### Organize flags by capability
+
+Group related flags together to make them easier to manage:
+
+```ts
+export const flagSchema = z.object({
+  // One group per capability
+  ticketClassification: z.object({
+    model: z.string().default('gpt-4o-mini'),
+    temperature: z.number().default(0.7),
+  }),
+  
+  emailGeneration: z.object({
+    model: z.string().default('gpt-4o'),
+    tone: z.enum(['formal', 'casual']).default('formal'),
+  }),
+  
+  documentRetrieval: z.object({
+    topK: z.number().default(5),
+    similarityThreshold: z.number().default(0.7),
+  }),
+});
+```
+
+### Set sensible defaults
+
+Choose defaults that work well for most cases. Experiments then test variations:
+
+```ts
+ticketClassification: z.object({
+  // Default to the fastest, cheapest model
+  model: z.string().default('gpt-4o-mini'),
+  
+  // Default to moderate temperature
+  temperature: z.number().min(0).max(2).default(0.7),
+  
+  // Default to the approach that works best in production
+  systemPrompt: z.enum(['concise', 'detailed']).default('concise'),
+}),
+```
+
+### Use enums for discrete choices
+
+When flags have a fixed set of valid values, use enums:
+
+```ts
+// Good: type-safe, clear options
+tone: z.enum(['formal', 'casual', 'friendly']).default('formal'),
```

Comment:
this is also important for something like `model` where passing `string` will type error with the ai sdk